File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length==0){
            return "";
        }
        String temp=strs[0];
        
        for(int i=1; i<strs.length;i++){
            for(int j=0;j<temp.length();j++){
                if(j==strs[i].length() || temp.charAt(j)!=strs[i].charAt(j)){
                    temp=temp.substring(0,j);
                    break;
                }
            }
        }
        return temp;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs[0]==""){ return ""; }
        if(strs.size()==1) return strs[0];
        string result;
        char ch;
        bool flag=false;
        
        for(int i=0; i<strs[0].size(); i++){
            for(int j=1; j<strs.size(); j++){
                ch=strs[0][i];
                if(ch!=strs[j][i]){
                    flag=true;
                }
            }
            if(!flag){ result.push_back(ch); }
        }
        return result;
    }
};